middleman-core/lib/middleman-core/renderers/haml.rb

```diff
@@ -52,6 +52,10 @@ def self.render_with_options(text, compiler_options)
               modified_options = options.dup
               modified_options[:context] = compiler_options[:context]
 
+              renderer_symbol = self.to_s.split(":").last.downcase.to_sym
```

Comment:
I agree this is a nasty but I'm fairly unfamiliar with what's available for introspection here and I didn't want to make the patch too Markdown-specific since this is a loop for multiple renderers.